src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateObject = [];
+
+  let currentState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    const { type, extraData, keysToRemove } = actions[i];
```

Comment:
for..of loop works better here

```suggestion
  for (const  { type, extraData, keysToRemove } of actions) {
   ...
```